Multiple Schema Support

amerryma opened the pull request:
1. Migra will now support multiple schemas from cli.

Example:

```
migra --schema app_public,app_private postgresql://postgres@localhost/graphile_starter postgresql://postgres@localhost/arid
```

2. Added domain support. You can test this by adding a domain in one db, then comparing a db without the domain. This should be an idempotent "if exists" call.

3. Added idempotent permission revocation. Use an example grant: https://www.postgresql.org/docs/9.0/sql-grant.html to see it work.

Run tests.

amerryma committed:
Added multiple schema support

amerryma committed:
Remove migra as a dev dependency

amerryma committed:
Added test :)

amerryma committed:
Added idempotent domain support
Allowed domain checking to not worry about visibility checks
Added idempotent revoking of permissions.

sean-schaefer commented on pyproject.toml:
```diff
@@ -22,7 +22,7 @@ pytest-clarity = {version=">=0.3.0-alpha.0", python=">=3.5,<4"}
 flake8 = "*"
 isort = {version=">=5", python=">=3.6,<4"}
-migra = "*"
+#migra = "*"
```
We should add a comment here.

sean-schaefer commented on schemainspect/pg/sql/domains.sql:
```diff
@@ -24,6 +24,5 @@ FROM pg_catalog.pg_type t
 WHERE t.typtype = 'd'
       AND n.nspname <> 'pg_catalog'
       AND n.nspname <> 'information_schema'
-  AND pg_catalog.pg_type_is_visible(t.oid)
```
Can you explain this change to me?

amerryma replied:
I was mentioning this the other day, but basically depending what you set your search path to, it will resolve differently. We don't really have any control to the search path that is set by migra when running the visibility checks to see if this domain exists, especially because it is running across multiple schemas. Read more here: https://www.postgresql.org/docs/12/ddl-schemas.html#DDL-SCHEMAS-PATH

At the end of the day, I don't care if it's visible within a search path because we always specify the schema along with the table name. And when comparing db diffs, we shouldn't care about if it was visible to the person querying, just that it exists.

sean-schaefer commented on tests/test_privileges.py:
```diff
@@ -18,8 +18,20 @@ def test_inspected_privilege():
         b2.create_statement
         == 'grant modify on function "schema"."test_function" to "test_user";'
     )
+    print(b.drop_statement)
```
Is this print still needed?

sean-schaefer commented on schemainspect/pg/obj.py:
```diff
@@ -913,7 +913,20 @@ def quoted_target_user(self):
     def drop_statement(self):
-        return "revoke {} on {} {} from {};".format(
+        return """DO
+$$
```
There's a failing test on line 454 of `test_all.py` that seems to be due to these changes -- it's still expecting the base SQL rather than the PL/pgSQL version.

sean-schaefer reviewed (changes requested):
Generally it seems to be working, though I did have to do the following to run the tests:
``` sql
CREATE USER cancer;
CREATE DATABASE cancer;
ALTER USER cancer WITH superuser;
```
We should probably document this somewhere. Btw I ran the tests with `make test`.

amerryma committed:
Fixed tests.
Added github action
I really don't know why we are using migra at all as a dev dep so we remove it.

amerryma committed:
Removing pip

amerryma committed:
Install poetry

amerryma committed:
Added back flake8

sean-schaefer committed:
Don't publish to PyPi in CI

sean-schaefer committed:
Update apt before trying installs

sean-schaefer committed:
Remove Github actions in favor of CircleCI

sean-schaefer committed:
Switch from pytest to make test

sean-schaefer reviewed (approved):
All good
